### ISSUE TITLE
Masterbar: add Actionbar on front-end

### ISSUE THREAD

**Issue by jeherve** [Automattic/jetpack#6848]
WordPress.com includes an action bar at the bottom right of each site. It gives you quick access to things like the customizer, the link to edit the post, the post's stats, and more:

![screen shot 2017-03-31 at 19 28 23](https://cloud.githubusercontent.com/assets/426388/24561753/56f8d7aa-1648-11e7-9b2c-14d25caaa9f8.png)
![screen shot 2017-03-31 at 19 28 57](https://cloud.githubusercontent.com/assets/426388/24561759/58ca4c62-1648-11e7-819f-835d03693054.png)

As we introduce the masterbar in Jetpack and thus remove some of the quick action buttons from the core WordPress admin bar, it becomes more and more important to ship the action bar in Jetpack.

**Comment by jenhooks** [comment]
More feedback prompted by a user in 3134271-t. Without the WordPress.com toolbar activated, users can click on "Edit Post" in the admin bar to go back to their post:

<img width="798" alt="editpost_option" src="https://cloud.githubusercontent.com/assets/7585054/25454088/49956a8c-2a99-11e7-894c-e8ddde919eb9.png">

With the WordPress.com toolbar activated, that option is no longer there:

<img width="894" alt="editpost_option2" src="https://cloud.githubusercontent.com/assets/7585054/25454202/9ffd6f28-2a99-11e7-8e0f-260c0976ab25.png">

Users will inevitably miss this link as well as others (Updates, Customize, Comments), and if we are truly interested in 'centralizing the WordPress experience with a single global toolbar' we certainly shouldn't be removing useful functionality.

**Comment by tmmbecker** [comment]
Additional feedback from an AT user in 614730-zen. 

```
My reason for deactivating is I need a toolbar with an 'Edit' option on it - so 
when I am viewing my site pages, I can quickly access the page to edit it, 
without having to navigate through the Pages listings in Wordpress.

Your toolbar has a 'write' button, but not an 'edit' button.

Can you help me disable the toolbar, or else update your toolbar to include an 
'edit' button?
```
I'll add, that I keep the WordPress.com toolbar disabled because of this same missing feature.

**Comment by lancewillett** [comment]
Anecdotal feedback from real life usage, on a Jetpack site I have for personal notes. (Also dogfooding o2+P2 in a non-a8c environment!)

I really miss the edit from front capability that the actionbar brings — even more than anything else. Because it's P2, the post and page "edit" links don't lead to Calypso, so I have a really hard time getting to the editor from the front-end.

I find myself scrolling down and mousing around the right corner of the screen a lot. I miss the floating actions there.

**Comment by dougaitken** [comment]
Driveby comment to add a business plan user in a Slack group I'm in asked:

> Why is there no edit page/post button in the top menu bar on WP.com like the one in WP.org?
> That button was always so convenient

**Comment by KokkieH** [comment]
Another request in 894741-f

**Comment by rachelmcr** [comment]
Another request came in 1063926-zen.

**Comment by jeherve** [comment]
Also in 1135359-zen

**Comment by jeherve** [comment]
Also in 1135591-zen

**Comment by liviopv** [comment]
Also 1271516-zen

**Comment by coder-karen** [comment]
Requested in 1343653-zen

**Comment by tmmbecker** [comment]
requested by a very frustrated AT user, "its two steps back to do anything in the front end" in 6269747-h

**Comment by ehti** [comment]
@jeherve  is there anyone working on this? I just got a super frustrated Business plan user.

**Comment by jeherve** [comment]
@ehti No, no one is working on this right now, but we'll update this issue once someone starts working on that.

**Comment by lancewillett** [comment]
Thanks @ehti for raising awareness of this one again. It's still a gap across the end-to-end experience (regardless of the underlying technology, which probably only we as technical folks care about).

CC @beaulebens @apeatling @catehstn as an example of a long-standing UX issue we should make a quick decision on soon. Either bring parity to Jetpack sites, or close the request and clearly state why we won't be launching this.

**Comment by dcoleonline** [comment]
Another request in 197293-zen

**Comment by sdixon194** [comment]
Another request in 2018012-zem

**Comment by tmmbecker** [comment]
2092902-zen

**Comment by iamgabrielma** [comment]
Another request on 14378607-hc

They're looking for the post short link option that shows on https://en.support.wordpress.com/shortlinks/ but this was not available for them

**Comment by jeremeylduvall** [comment]
@jeherve We had a VIP client raise this as an issue in 100956-z. Per @lancewillett comment above:

> Either bring parity to Jetpack sites, or close the request and clearly state why we won't be launching this.

Are you aware of any more recent discussion on the topic?

**Comment by jeherve** [comment]
> Are you aware of any more recent discussion on the topic?

We've made no progress on this yet, I don't think it's been discussed anywhere recently.

**Comment by sophiegyo** [comment]
I chatted with someone who was also interested in this: 4532528-hc

**Comment by Aurorum** [comment]
7866689-zen

**Comment by jeherve** [comment]
Closing as duplicate of https://github.com/Automattic/wp-calypso/issues/36795
